mdp/nodes/sfa_nodes.py

```diff
@@ -1,12 +1,25 @@
-from builtins import str
+
+#====================================================================[ header ]
+
 from builtins import range
 __docformat__ = "restructuredtext en"
 
+import numpy as np
```

Comment:
I realize that `numx` is solving a problem that does not exist anymore, but either you remove it completely everywhere (in another PR), or you keep using it instead of `np`. It does not hurt.

Reply:
We already had a case where we ran into a function that behaved differently from the numpy-default we expected. I'd switch the whole thing to straight up numpy; maybe I can find the time to do it.

Reply:
Well, no, the idea of `mdp.numx` was to be able to use `numpy` or `scipy` transparently. I.e. if you have only `numpy` installed, then fine, if you have `scipy` use the "better" routines available there. Note that even @Stewori noticed that the SVD implementation is way faster in `scipy` than in `numpy`. The very same arguments holds for most of the linear algebra routines in `numpy`/`scipy`. What may be obsolete is the idea of not requiring `scipy` by default. This would mean make `scipy` a hard dependency. I am fine with that, if you think it's worth it. But note that `numpy` is practically available pre-installed by default everywhere, so if you don't care about performance you can now use MDP without pulling in all of `scipy`. But, again, YMMV…

Reply:
That makes a lot of sense. However, we already had one case where a `scipy`method differed enough from its `numpy` equivalent to cause erroneous behavior. I'd definitely be in favor of picking one or the other.
Making `scipy` a hard dependency for a machine learning library should be fine?

Reply:
What we are doing right now can be understood in `mdp/utils/__init__.py`. We wrote a little compatibility layer between `numpy` and `scipy` queivalent functions, so that we are sure we use internally a consistent interface. We have `mdp.utils.svd` for example. Look also at the casting issues we address there (`numpy` casts to double even if the inputs are single in several functions). If the difference in results between a `numpy` function and a `scipy` one are more than just cosmetic, then this is bug in `numpy` or `scipy` and needs to be reported there, I think.

Reply:
Right, I don't remember the details of what exactly the problem was that we had, so not sure how much of an issue this actually is. Changing the code to explicitly use either `numpy` or `scipy` would be subject to another PR anyway, so I'll just change it back to `numx` and we can think about changing this later should we feel it's necessary.